include/amici/defines.h

```diff
@@ -91,7 +91,7 @@ typedef enum AMICI_sensi_meth_TAG {
 } AMICI_sensi_meth;
 
 /** linear solvers for CVODES/IDAS */
-enum LinearSolver {
+enum class LinearSolver {
```

Comment:
I like using enum class.

Reply:
I was not sure, but we should also be able to replace the typedef enums right?

Reply:
Yes, they can also be replaced. That's from back in C time.
Since the will be always specified with the class qualifier, we can also rid of any common prefixes.
By convention, enum class member names are not capitalized. Just as comment, we can keep the existing ones the way they are.